pkg/controller/keystone/keystone_controller.go

```diff
@@ -145,7 +146,8 @@ func (r *ReconcileKeystone) Reconcile(request reconcile.Request) (reconcile.Resu
 		if ready {
 			return reconcile.Result{}, r.updateStatusWithExternalKeystone(keystone)
 		}
-		return reconcile.Result{}, err
+		reqLogger.Info(err.Error())
+		return reconcile.Result{RequeueAfter: time.Duration(keystone.Spec.ServiceConfiguration.ExternalAddressRetrySec) * time.Second}, nil
```

Comment:
Should we return err here?

Reply:
In case of cc, operator is deployed first and it waits until keystone will be deployed on openstack cluster. It will be expected behavior to not have keystone available to us on the beginning, that is why it is checked here and not treated as an error. Returning error could change the time after which next reconcile loop iteration happens, some resources claims it should trigger reconcile immediately on which previous approach based on, but from my observation it behaves differently.